app/src/main/java/com/babylon/wallet/android/RadixApplication.kt

```diff
@@ -54,19 +58,22 @@ class RadixApplication : Application(), Configuration.Provider {
 
         appsFlyerIntegrationManager.init()
         bootstrapHomeCards()
-        ProcessLifecycleOwner.get().lifecycle.addObserver(AppLifecycleObserver(appLockStateProvider, scope))
+        ProcessLifecycleOwner.get().lifecycle.addObserver(AppLifecycleObserver())
     }
 
-    class AppLifecycleObserver(
-        private val appLockStateProvider: AppLockStateProvider,
-        private val scope: CoroutineScope
-    ) : DefaultLifecycleObserver {
+    inner class AppLifecycleObserver : DefaultLifecycleObserver {
+
+        override fun onStart(owner: LifecycleOwner) {
+            super.onStart(owner)
+            accountLockersObserver.startMonitoring()
+        }
 
         override fun onStop(owner: LifecycleOwner) {
             scope.launch {
                 appLockStateProvider.lockApp()
             }
             super.onStop(owner)
+            accountLockersObserver.stopMonitoring()
```

Comment:
Right, I'd exclude WalletScreen because I think it should hold (as much as possible) only the logic related to the screen itself (account cards, total balance, etc) and not something totally unrelated to that. But we definitely could use MainActivity instead of the App class since we have a single activity and it will result in the same outcome. The app class in my perception is the more "correct" place because it indicates that the observer is registered for the whole app rather than a specific activity of the app. But as I said it would have resulted in the same outcome since we have a single activity.